p2p/server/server.go

```diff
@@ -188,7 +187,7 @@ func (s *Server) queueHandler(ctx context.Context, stream network.Stream) {
 	_ = stream.SetDeadline(time.Now().Add(s.timeout))
 	defer stream.SetDeadline(time.Time{})
 	rd := bufio.NewReader(stream)
-	size, err := varint.ReadUvarint(rd)
+	size, err := binary.ReadUvarint(rd)
```

Comment:
please doublecheck that binary.ReadUvarint behaves identically with varint.ReadUvarint. application will fail with random errors otherwise

Reply:
There used to be a bug in `binary.ReadUvarint` that was fixed with Go 1.20: https://go.dev/doc/go1.20#encodingbinarypkgencodingbinary

The only difference between `varint.ReadUvarint` and `binary.Readvarint` are that the first returns an error if a `uint` is not encoded minimally, e.g. `0x81 0x00` is decoded by the standard library to `1`, while `varint.ReadUvarint` would `varint.ErrNotMinimal` because `1` can be more efficiently encoded as `0x01`.

I revert this PR and create a new one without the change to be safe.